src/brevitas/graph/equalize.py

```diff
@@ -245,55 +331,100 @@ def _cross_layer_equalization(
+        else:
+            return torch.tensor(1., dtype=dtype, device=device)
+        # If there is a mismatch in the activation channel (e.g. a transpose/flatten op in between),
+        # do not perform equalization
+        if any([act_axis != axis for axis in list_of_source_axes + list_of_sink_axes]):
+            return torch.tensor(1., dtype=dtype, device=device)
 
     # Check if any of the axis is None, which means that the module is not supported.
     # In that case, do not perform graph equalization
-    if None in [*src_axes.values(), *sink_axes.values()]:
+    axes_to_check = [*src_axes.values(), *sink_axes.values()]
+    if None in axes_to_check:
         return torch.tensor(1., dtype=dtype, device=device)
 
-    src_size = _get_size(src_axes)
+    # Check if the sink_size is None,
+    # which means that the some of the sinks do not have the same size as the others.
     sink_size = _get_size(sink_axes)
-
-    # Check if any of the src_size or sink_size is None, which means that the some of the
-    # sources or sinks do not have the same size as the others.
-    # Similarly, exit if source and sink have different different sizes
-    if None in [src_size, sink_size] or src_size != sink_size:
+    if None in [sink_size]:
         return torch.tensor(1., dtype=dtype, device=device)
-    transpose = lambda module, axis: module.weight if axis == 0 else module.weight.transpose(0, 1)
+
     scale_fn = _select_scale_computation_fn(scale_computation_type)
-    if merge_bias:
-        src_weights = [
-            _combine_weights_bias(transpose(m, axis), bias_shrinkage, m.bias) for m,
-            axis in src_axes.items()]
-    else:
-        src_weights = [transpose(m, axis) for m, axis in src_axes.items()]
     sink_weights = [transpose(m, axis) for m, axis in sink_axes.items()]
-    srcs_range = scale_fn(torch.cat([w.reshape(w.size(0), -1) for w in src_weights], 1))
     sinks_range = scale_fn(torch.cat([w.reshape(w.size(0), -1) for w in sink_weights], 1))
-    sinks_range += EPSILON
+    sinks_range = torch.clamp(sinks_range, EPSILON)
+
+    # Determine the srcs_range based on where we are performing activation equalization or
+    # weight equalization
+    if list_of_act_val is not None:
+        act_shape = list_of_act_val[0].shape
+        # act_val = [WeightBiasTuple(act_val) for act_val in list_of_act_val]
+        list_of_act_val = [
+            transpose(WeightBiasTuple(act_val), act_axis) for act_val in list_of_act_val]
+        srcs_range = scale_fn(
+            torch.cat([act_val.reshape(act_val.size(0), -1) for act_val in list_of_act_val],
+                      1))  # scale_fn(act_val.reshape(act_val.size(0), -1))
+    else:
+        # If we do weight equalization, perform additional check on source size
+        src_size = _get_size(src_axes)
+        # Exit if source and sink have different different sizes, or if sources contains None
+        if src_size != sink_size or None in [src_size]:
```

Comment:
Should we raise a warning on different sizes? Is that something that can happen during a correct execution?